include/execution.hpp

```diff
@@ -1468,6 +1465,184 @@ namespace std::execution {
 
   // NOT TO SPEC
   using manual_event_loop = __loop::manual_event_loop;
+
+  inline namespace __schedule_from {
+    namespace __impl {
+      template <class Fn>
+          requires is_nothrow_move_constructible_v<Fn>
+        struct __conv {
+          Fn __fn_;
+          operator invoke_result_t<Fn> () && {
+            return ((Fn&&) __fn_)();
+          }
+        };
+      template <class Fn>
+        __conv(Fn) -> __conv<Fn>;
+
+      template <class Scheduler_, class Sender_>
+        struct __sender {
+          using Scheduler = __t<Scheduler_>;
+          using Sender = __t<Sender_>;
+          Scheduler __sch_;
+          Sender __snd_;
+
+          template <class CvrefReceiver_>
+            struct __op1 {
+              // Bit of a hack here: the cvref qualification of the Sender
+              // is encoded in the type of the Receiver:
+              using CvrefReceiver = __t<CvrefReceiver_>;
+              using CvrefSender = __member_t<CvrefReceiver, Sender>;
+              using Receiver = decay_t<CvrefReceiver>;
+
+              // Compute a variant type that is capable of storing the results of the
+              // input sender when it completes. The variant has type:
+              //   variant<
+              //     tuple<set_done_t>,
+              //     tuple<set_value_t, decay_t<Values1>...>,
+              //     tuple<set_value_t, decay_t<Values2>...>,
+              //        ...
+              //     tuple<set_error_t, decay_t<Error1>>,
+              //     tuple<set_error_t, decay_t<Error2>>,
+              //        ...
+              //   >
+              template <class... Ts>
+                using __bind_tuples = __bind_front<variant, tuple<set_done_t>, Ts...>;
+
+              using __bound_values_t = typename sender_traits<Sender>::template value_types<
+                  __transform<
+                      __bind_front<tuple, set_value_t>::template __f,
+                      decay_t
+                  >::template __f,
+                  __bind_tuples>;
```

Comment:
@lewissbaker This causes the source sender's values to be decay-copied into the operation state. I think this is correct, but the current wording in P2300 doesn't reflect the fact that the values are decay-copied before being propagated to the completion sender.